File: 02_algorithm/programmers/Level4/숫자 블록.py
import math

def solution(begin, end):
    answer = []
    
    if begin == 1:
        answer.append(0)
        begin += 1
        
    for block_idx in range(begin, end + 1):
        for i in range(2, int(math.sqrt(block_idx)) + 1):
            mok = block_idx // i
            if mok > 10000000:
                continue
            if block_idx % i == 0:
                answer.append(mok)
                break
        else:
            answer.append(1)
            
    return answer


print(solution(1, 10))

File: 06_js/04_algorithm/programmers/Level4/숫자 블록.js
function solution(begin, end) {
  let answer = [];
  
  if (begin === 1) {
    answer.push(0);
    begin += 1;
  }
  
  for (let i = begin; i <= end; i++) {
    let checkBreak = false;
    let sqrtValue = parseInt(Math.sqrt(i));
    for (let j = 2; j <= sqrtValue; j++) {
      const mok = parseInt(i / j);
      if (mok > 10000000) {
        continue;
      }
      if (i % j === 0) {
        answer.push(mok);
        checkBreak = true;
        break;
      }
    }
    if (!checkBreak) {
      answer.push(1);
    }
  }
  return answer;
}

console.log(solution(1, 10));